src/supervisor/watchers/handlers/informer-config.ts

```diff
@@ -0,0 +1,149 @@
+import { ADD, DELETE, UPDATE } from '@kubernetes/client-node';
+
+import { WorkloadKind } from '../../types';
+import * as pod from './pod';
+import * as cronJob from './cron-job';
+import * as daemonSet from './daemon-set';
+import * as deployment from './deployment';
+import * as job from './job';
+import * as replicaSet from './replica-set';
+import * as replicationController from './replication-controller';
+import * as statefulSet from './stateful-set';
+import * as deploymentConfig from './deployment-config';
+import { IWorkloadWatchMetadata } from './types';
+
+/**
+ * This map is used in combination with the kubernetes-client Informer API
+ * to abstract which resources to watch, what their endpoint is, how to grab
+ * a list of the resources, and which watch actions to handle (e.g. a newly added resource).
+ *
+ * The Informer API is just a wrapper around Kubernetes watches that makes sure the watch
+ * gets restarted if it dies and it also efficiently tracks changes to the watched workloads
+ * by comparing their resourceVersion.
+ *
+ * The map is keyed by the "WorkloadKind" -- the type of resource we want to watch.
+ * Legal verbs for the "handlers" are pulled from '@kubernetes/client-node'. You can
+ * set a different handler for every verb.
+ * (e.g. ADD-ed workloads are processed differently than DELETE-d ones)
+ *
+ * The "listFunc" is a callback used by the kubernetes-client to grab the watched resource
+ * whenever Kubernetes fires a "workload changed" event and it uses the result to figure out
+ * if the workload actually changed (by inspecting the resourceVersion).
+ */
+export const workloadWatchMetadata: Readonly<IWorkloadWatchMetadata> = {
```

Comment:
I _feel_ like this object could be even more streamlined, there is a lot of repeated pattern. could you have a function that passes in the imported e.g. `pod` and returns the config? the names of the handlers would have to be standardised e.g. `podWatchHandler` to `watchHandler` - this would make adding one even easier!